### PR TITLE
Group development dependencies

### LINKED ISSUE #673: Group Renovate dependencies to limit test runs
### What feature would you like to have?

Adjust Renovate configuration so it will group common dependencies together.

| Grouping | Security critical | Update frequency |
| --- | --- | --- |
| Next.js, React (Framework stack) | yes | daily |
| ESLint, Prettier, ESLint plugins | not likely | monthly |
| Jest, Cypress, and testing utils | not likely |  monthly |
| Chakra UI, emotion, and Framer Motion | not likely | monthly |
| React Query + Octokit | not likely | monthly |
| Rehype + Remark ecosystem | not likely | monthly |
| TypeScript, TS types, and type-fest | not likely | monthly |
| lint-staged and husky | not likely | monthly |

### What problem will this feature solve?

Right now whenever we have a update PR, and the PR branch gets behind `main`, Renovate will "rebase" so that it's current with `main`. This causes the tests to run again.
It gets even worse when we have more than one update PR ready to go, because all other PR branches get "rebased" as well.

~We're going to use Percy to test for visual changes, but we do not get unlimited resources, so we will need to be smart about the amount of runs we do.~

By grouping updates into logical buckets, we can reduce the amount of PRs that Renovate will create. This will help reduce the amount of "rebase" actions that Renovate performs.

If we want a update ahead of schedule, we can request it from the Dependency Dashboard.

### Do you want to help make the feature?

Yes

### Is there anything else we need to know?

Relevant Renovate docs:

- https://docs.renovatebot.com/noise-reduction/#package-grouping
- https://docs.renovatebot.com/noise-reduction/#selective-scheduling

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Use the `npm:unpublishSafe` preset instead of having our own custom `packageRule`
- Group all development dependencies, and update them once a month
- Order the `packageRules` by importance

## Context

The `packageRules` config option docs say:

> Important to know: Renovate will evaluate all `packageRules` and not stop once it gets a first match. You should order your `packageRules` in order of importance so that later rules can override settings from earlier rules if needed.

Closes #673.

## Tests

I've checked my work by:

- [ ] Adding new tests or adjusting existing tests
- [ ] Testing manually
- [x] Code review

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -12,21 +13,26 @@
...
 	"packageRules": [
 		{
-			"description": "Wait 3 days before creating a npm update PR",
-			"matchDatasources": ["npm"],
-			"stabilityDays": 3
-		},
-		{
-			"description": "Automerge ESLint and Prettier updates",
+			"description": [
+				"Group all non-major yarn development dependencies together, get updates once a month"
```
Comment (Belco90) [inline renovate.json:17]: Why "yarn"?

**Review by Belco90 — COMMENTED** [review]
So if I understood this correctly: with the new config, the dev packages will take a month to be updated. Then I assume in another MR we will set up the critical packages to be updated more often (like Next.js or React). Is this correct?

File: renovate.json (hunk 1 above)
Comment (HonkingGoose) [reply renovate.json:17]:
```suggestion
				"Group all non-major pnpm development dependencies together, get updates once a month"
```

**Comment by HonkingGoose** [comment]
I can set up the critical packages in this PR as well! 😉

File: renovate.json [hunk 2]
```diff
@@ -12,21 +13,26 @@
...
-			"matchDatasources": ["npm"],
-			"stabilityDays": 3
-		},
-		{
-			"description": "Automerge ESLint and Prettier updates",
+			"description": [
+				"Group all non-major pnpm development dependencies together, get updates once a month"
+			],
+			"matchManagers": ["npm"],
 			"matchDepTypes": ["devDependencies"],
```
Comment (HonkingGoose) [inline renovate.json:20]: This is the key part: `matchDepTypes=devDependencies` makes it so this only groups `devDependencies` in the `package.json` file.

Our framework stack (Next.js and React) is in the `dependencies` part of the `package.json` file:

```json
	"dependencies": {
		"@chakra-ui/react": "2.2.8",
		"@emotion/react": "11.10.0",
		"@emotion/styled": "11.10.0",
		"@octokit/auth-callback": "3.0.0",
		"@octokit/rest": "19.0.3",
		"@sentry/nextjs": "7.10.0",
		"@tanstack/react-query": "4.1.0",
		"@tanstack/react-query-devtools": "4.0.10",
		"downshift": "6.1.9",
		"framer-motion": "7.1.0",
		"lodash-es": "4.17.21",
		"next": "12.2.5",
		"next-seo": "5.5.0",
		"nookies": "2.5.2",
		"react": "18.2.0",
		"react-dom": "18.2.0",
		"react-icons": "4.4.0",
		"rehype-highlight": "5.0.2",
		"rehype-react": "7.1.1",
		"remark-gfm": "3.0.1",
		"remark-github": "11.2.4",
		"remark-parse": "10.0.1",
		"remark-rehype": "10.1.0",
		"remark-stringify": "10.0.2",
		"semver": "7.3.7",
		"unified": "10.1.2"
	},
```

So these `dependencies` are _not_ affected by the new grouping. We'll get separate PRs for each of these `dependencies` when they're three days old.

Comment (Belco90) [reply renovate.json:20]: Great! Thanks for the clarification.

**Review by Belco90 — APPROVED** [review]

**Pull request merged**